controllers/inferenceservice_controller.go

```diff
@@ -159,8 +159,13 @@ func (r *OpenshiftInferenceServiceReconciler) SetupWithManager(mgr ctrl.Manager)
 	}
 
 	if isAuthorinoRequired {
-		builder.Owns(&authorinov1beta2.AuthConfig{})
-		r.log.Info("kserve is enabled with Service Mesh, Authorino is a requirement")
+		r.log.Info("kserve is enabled with Service Mesh")
+		if utils.IsGroupRegisteredInScheme(r.client, authorinov1beta2.GroupVersion) {
+			r.log.Info("Authorino is registered, enabling Authorization capability")
+			builder.Owns(&authorinov1beta2.AuthConfig{})
+		} else {
```

Comment:
If Knative/Istio are installed on the cluster but Authorino is not installed, the odh-model-controller do not enable the authConfig reconciler, as it relies on Authorino for certain authentication functionalities. While this log shows "Authorino capability disabled", it's important to inform the user the warning message when the user attempts to enable authentication but Authorino is not installed. It should clearly explain that Authorino is required for certain authentication functionalities.

What do you think?

Reply:
by doing this, even though KServe requires it and authorino is missing, it will not require it leading to issuer later, right?

Reply:
I changed the logic, PTAL at new version